week5/src/pages/ContentPage.tsx

```diff
@@ -0,0 +1,80 @@
+import { useEffect, useState } from 'react';
+import { useParams } from 'react-router-dom';
+import styled from 'styled-components';
+import { StyledInput } from '../components/Input';
+import Layout from '../components/Layout';
+import { useGetLetterHook } from '../hooks/useGetLetterHook';
+
+const ContentPage = () => {
+  const [input, setInput] = useState('');
+  const [IntLetterId, setIntLetterId] = useState<number>(0);
+  const [check, setCheck] = useState<boolean>(false);
+  const { letterId } = useParams();
+  const { letters } = useGetLetterHook();
+
+  useEffect(() => {
+    if (letterId === undefined) return;
+    setIntLetterId(parseInt(letterId));
+  }, [letterId]);
```

Comment:
이게 진짜 맞을까요!

Reply:
```suggestion ts
  useEffect(() => {
    if (letterId)   setIntLetterId(parseInt(letterId));
  }, [letterId]);

```
요러케만 하면덴다! 이게 어쩔수없이 parseInt라는 함수안에 undefined가 들어가면 안데서! 한번 체크를 해주는게 마자!! 다만 undefined일때는 아무것도 실행안하면 데니까 우리는 letterId의 값이 있는지 확인해서 set을 진행하면 되는고야!✨

Reply:
추가로 만약 이과정이 불편하면 url에서 가져오는거 말고  location 사용하는 법도 있어!
```typescript
//이렇게 보내고
navigate(`/content`, {state :{letterId : idx+""});

// 아래처럼 받으면 useEffect필요업숴!
interface LetterIdLocation {
  state: {letterId: string};
}
const navigate = useNavigate();
 const { state } = useLocation() as LetterIdLocation;

```